n2kMapper.js

```diff
@@ -15,33 +15,55 @@ function N2kMapper (options) {
   this.state = {}
 
   this.on('n2kRequestMetadata', (src) => {
-    this.emit('n2kOut', {
-      "pgn": 59904,
-      "dst": src,
-      "PGN": 126996
-    })
-    this.emit('n2kOut', {
-      "pgn": 59904,
-      "dst": src,
-      "PGN": 126998
-    })
-    this.emit('n2kOut', {
-      "pgn": 59904,
-      "dst": src,
-      "PGN": 60928
-    })
+    if ( src === 255 ) {
+      return
+    }
+    this.requestMetaData(src, 126996)
+    this.requestMetaData(src, 126998)
+    this.requestMetaData(src, 60928)
   })
 }
 
+N2kMapper.prototype.requestMetaData = function(src, pgn) {
+  const reqPgn = {
+    "pgn": 59904,
+    "dst": src,
+    "PGN": pgn
+  }
+  let retries = 5
+  debug(`requesting pgn ${pgn} from src ${src}`)
+  this.emit('n2kOut', reqPgn)
+  let interval = setInterval(() => {
+    if ( retries-- === 0 ) {
+      debug(`did not get meta pgn ${pgn} for src ${src}`)
+      clearInterval(interval)
+    } else if ( this.state[src] && this.state[src].metaPGNsReceived && this.state[src].metaPGNsReceived.indexOf(pgn) != -1 ) {
+      clearInterval(interval)
+      debug(`got meta pgn ${pgn} from src ${src}`)
+    } else {
+      debug(`did not get pgn ${pgn} from src ${src}, retrying..`)
+      this.emit('n2kOut', reqPgn)
+    }
+  }, 2000)
```

Comment:
Is a fixed delay good or should it increase per retry?

Reply:
I think fixed is ok, but I'm also ok with changing it...